src/App.js

```diff
@@ -1,16 +1,136 @@
 import React from 'react';
+import './App.css';
+import ToDoList from './components/TodoComponents/TodoList';
+import Form from './components/TodoComponents/TodoForm';
+
+
+// you will need a place to store your state in this component.
+// design `App` to be the parent component of your application.
+// this component is going to take care of state, and any change handlers you need to work with your state
+
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor(props) {
+    super(props);
+    this.state = {
+      list: [
+        {
+          task: 'Check The Mail',
+          id: 152881707,
+          done: false,
+          completed: 'Incomplete',
+          icon: "fas fa-check-circle red",
+        },
+        {
+          task: 'Workout',
+          id: 152881708,
+          done: false,
+          completed: 'Incomplete',
+          icon: "fas fa-check-circle red",
+        }, 
+        {
+          task: 'Walk the dogs',
+          id: 152881709,
+          done: false,
+          completed: 'Incomplete',
+          icon: "fas fa-check-circle red",
+        }
+      ],
+      inputText: '',
+      newTask: '',
+    }
+    }// the three tasks are auto added though can be handled just like every task the user adds
+
+  completeTask = (index) => {
+      this.setState({
+        list: this.state.list.map((task) => {
+
+          if (index !== task.id) {
+            return task;
+          } else {
+            return {
+              ...task,
+              completed: task.completed === 'Complete' ? 'Incomplete' : 'Complete',
+              icon: task.icon === 'fas fa-check-circle green' ? 'fas fa-check-circle red' : 'fas fa-check-circle green'
+            };
+          }
+        })
+      });
+    };// gives user the ability to toggle the complete/incomplete event will set state function the line through and change the icons
+
+  closeTask = (index) => {
+      this.setState({
+        list: this.state.list.filter((task) => {
+
+          if (index !== task.id) {
+            return task;
+          } else {
+            if (task.completed === 'Complete') {
+              return null;
+            } else {
+              alert('please complete task');
+              return task;
+            }
+          }
+        })
+      }
+    )
+    }// pops up an alert when you try to delete a task that is not done
+
+  changeHandler = event => {
+    event.preventDefault();
+    this.setState({ newTask: event.target.value });
```

Comment:
remember to make this dynamic so you can use it on more then one input in the future.
```suggestion
    this.setState({ [event.target.name]: event.target.value });
```